pkmodel/solution.py

```diff
@@ -12,6 +18,182 @@ class Solution:
         an example paramter
 
     """
-    def __init__(self, value=44):
-        self.value = value
+    def __init__(self, dose_dataframe,model_instance):
+        self.dose_dataframe = dose_dataframe
+        self.model_instance = model_instance
 
+    def dose(self,t,row_index):
+        t = int(t)
+        if self.dose_dataframe.iloc[row_index]['tstart'] <= t <= self.dose_dataframe.iloc[row_index]['tend'] and self.dose_dataframe.iloc[row_index]['inst'] == 0 :#
+            dose = self.dose_dataframe.iloc[row_index]['doses']
+        else:
+            dose = 0
+        return dose
+
+    def solve_dataframe(self, absorb, comp, ):
+
+        if  comp == 0 and absorb == 0:
+            y0 = [0]
+        elif comp == 0 and absorb == 1: 
+            y0 = [0,0]
+        elif comp == 1 and absorb == 0:
+            y0 = [0,0]
+        elif comp == 1 and absorb == 1: 
+            y0 = [0,0,0]
+        elif comp == 2 and absorb == 0:
+            y0 = [0,0,0]
+        elif comp == 2 and absorb == 1: 
+            y0 = [0,0,0,0]
+        else :
+            raise("Invalid parameter values")
+
+        sol_dataframe = []
+
+        for i in range(self.dose_dataframe.shape[0]):
+            if self.dose_dataframe.iloc[i][3] == 1:
+                y0[0] += self.dose_dataframe.iloc[i][2]
+                
+
+            sol_row = self.solver(self.dose_dataframe, i, y0)
+            sol_y_flattened = sol_row.y.flatten()
+            y0 = sol_y_flattened[len(sol_row.y[0])-1::len(sol_row.y[0])]
+            sol_dataframe.append(sol_row)
+        
+        return sol_dataframe
+
+        
+    
+    def solver(self, dose_df, row_index, y0):
+        arguments = [
```

Comment:
Explain what this function does. What are the different cases? What do we check?